### PR TITLE
Oom fix

### PR DESCRIPTION
**Pull request by nithinraok**
* duration is not handled properly while reading segment of file 
* edited to check for int rather than check for None with int for offset variable

### CODE REVIEW

**Review by titu1994 — COMMENTED** [review]
Make the if case also update duration

**Comment by nithinraok** [comment]
> Make the if case also update duration
if case is required for offset since it is set to None if not found, its not the case with duration 
https://github.com/NVIDIA/NeMo/blob/11aaa8d8e69e2ce23f21b54f0bbc5b9b288df11b/nemo/collections/asr/parts/collections.py#L318

**Comment by titu1994** [comment]
Inside the tarred if case, duration is by default 0 (which should be None, not 0) as set at the top, and not updated. 

But the manifest still has duration no? So just as below in the else case, check if duration exists in the manifest file and extract it (else set it to None)

File: nemo/collections/asr/parts/perturb.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
 
 
 def read_one_audiosegment(manifest, target_sr, rng, tarred_audio=False, audio_dataset=None):
+    duration = 0
```
Comment (titu1994) [inline nemo/collections/asr/parts/perturb.py:64]: This should be None (which is the default for AudioSegment I believe)

File: nemo/collections/asr/parts/perturb.py [hunk 2]
```diff
@@ -74,9 +75,10 @@ def read_one_audiosegment(manifest, target_sr, rng, tarred_audio=False, audio_da
     else:
```
Comment (titu1994) [inline nemo/collections/asr/parts/perturb.py:75]: Above here, in the if case, update the value of duration from the manifest entry

**Review by titu1994 — CHANGES_REQUESTED** [review]

File: nemo/collections/asr/parts/perturb.py (hunk 2 above)
Comment (nithinraok) [reply nemo/collections/asr/parts/perturb.py:75]: yes, this I missed, added now for tarred dataset case

Comment (titu1994) [reply nemo/collections/asr/parts/perturb.py:75]: Looks good. Let @jbalam-nv review and then merge when ready.

**Review by titu1994 — COMMENTED** [review]
LGTM

File: nemo/collections/asr/parts/perturb.py (hunk 2 above)
Comment (jbalam-nv) [reply nemo/collections/asr/parts/perturb.py:75]: @nithinraok so this was causing your OOM errors? In my tests, I did not use duration as I split the noise files into 20s chunks. Looks good. Could you add a test for this :).

**Review by jbalam-nv — APPROVED** [review]

**Comment by jbalam-nv** [comment]
LGTM

**Comment by titu1994** [comment]
Tests exist for perturbatiojs (I think). If we haha one for noise Perturbation, could you add this to it ? Just a duration test of 1 second for non Tarred dataset is fine for now (measure that 1 second duration gives you 16000 vector length and that this vector does not match when you apply offset of 1 second). 

If the test doesn't exist, we can add it later.

File: nemo/collections/asr/parts/perturb.py (hunk 2 above)
Comment (nithinraok) [reply nemo/collections/asr/parts/perturb.py:75]: yes with these changes, runs looking good for now :)

**Comment by titu1994** [comment]
Seems we dont have tests for checking this. @jbalam-nv We can look into adding tests in a future PR (cause it deals with tarred and non-tarred data during tests, which is non-trivial to setup at the moment). For now, I believe it looks correct.

**Pull request merged**